samples/userspace/prod_consumer/README

```diff
@@ -0,0 +1,93 @@
+Producer/consumer Sample application that exercises some user mode concepts
```

Comment:
Please make this an RST file so it can be part of the documentation.

Reply:
done